src/sth_database.js

```diff
@@ -169,52 +175,117 @@
    * @return {string} The database name
    */
   function getDatabase(service) {
-    return sthConfig.DB_PREFIX + '_' + service;
+    return sthConfig.DB_PREFIX + service;
   }
 
   /**
    * Return the name of the collection which will store the raw events
+   * @param {string} databaseName The database name
    * @param {string} servicePath The service path of the entity the event is related to
    * @param {string} entityId The entity id related to the event
    * @param {string} entityType The type of entity related to the event
    * @param {string} attrName The attribute id related to the event
    * @returns {string} The collection name
    */
-  function getCollectionName4Events(servicePath, entityId, entityType, attrName) {
+  function getCollectionName4Events(databaseName, servicePath, entityId, entityType, attrName) {
+    var collectionName4Events;
     switch(sthConfig.DATA_MODEL) {
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_SERVICE_PATH:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath;
+        collectionName4Events = servicePath;
+        break;
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_ENTITY:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath + '_' + entityId + (entityType ? '_' + entityType : '');
+        collectionName4Events =  servicePath + '_' + entityId + (entityType ? '_' + entityType : '');
+        break;
       case sthConfig.DATA_MODELS.COLLECTIONS_PER_ATTRIBUTE:
-        return sthConfig.COLLECTION_PREFIX + '_' + servicePath + '_' + entityId + (entityType ? '_' + entityType : '') +
+        collectionName4Events =  servicePath + '_' + entityId + (entityType ? '_' + entityType : '') +
           '_' + attrName;
+        break;
     }
+    if (sthConfig.SHOULD_HASH) {
+      var limit = getHashSizeInBytes(databaseName);
+      if (limit < MIN_HASH_SIZE_IN_BYTES) {
+        sthLogger.warn('The available bytes for the hashes to be used as part of the collection names is not big enough (' +
+          'at least ' + MIN_HASH_SIZE_IN_BYTES + ' bytes are needed), ' +
+          'please reduce the size of the DB_PREFIX ("' + sthConfig.DB_PREFIX + '" = ' + bytesCounter.count(sthConfig.DB_PREFIX) + ' bytes), ' +
+          'the service ("' + databaseName.substring(sthConfig.DB_PREFIX.length, databaseName.length) + '" = ' + bytesCounter.count(databaseName) +
+          ' bytes) and/or the COLLECTION_PREFIX ("' + sthConfig.COLLECTION_PREFIX + '" = ' + bytesCounter.count(sthConfig.COLLECTION_PREFIX) +
+          ' bytes)',
+          {
+            operationType: sthConfig.OPERATION_TYPE.DB_LOG
+          }
+        );
+        return null;
+      }
+      return sthConfig.COLLECTION_PREFIX + generateHash(collectionName4Events, limit);
+    } else {
+      return sthConfig.COLLECTION_PREFIX + collectionName4Events;
```

Comment:
In fact, I just did it since it was straight forward until we decide about it :)

Currently, for the hashing case, the following message is logged in case the bytes available for the hash are less than 20 bytes:
```
time=2015-05-22T07:45:16.403Z | lvl=WARN | corr=NA | trans=NA | op=OPER_STH_DB_LOG | msg=The available bytes for the hashes to be used as part of the collection names are not enough (currently 17 and at least 20 bytes are needed), please reduce the size of the DB_PREFIX ("sth_" = 4 bytes), the service ("orion" = 5 bytes) and/or the COLLECTION_PREFIX ("ññññññññññññññññññññññññññññññññññññññññññññ" = 88 bytes) to save more bytes for the hash
```
On the other hand and for the other case, the following message is logged in case the namespace (concatenation of the database name and the collection name for the aggregated data is bigger than 120 bytes):
```
time=2015-05-22T07:42:31.101Z | lvl=WARN | corr=NA | trans=NA | op=OPER_STH_DB_LOG | msg=The size in bytes of the namespace for storing the aggregated data ("sth_orion" plus "ñññññññññññññññññññññññññññññññññññññññññññ/_entityId_entityType.aggr", 121 bytes) is bigger than 120 bytes, please reduce the size of the DB_PREFIX ("sth_" = 4 bytes), the service ("orion" = 5 bytes), the COLLECTION_PREFIX ("ñññññññññññññññññññññññññññññññññññññññññññ" = 86 bytes), the entity id ("entityId" = 8 bytes) and/or the entity type ("entityType = 10 bytes) to make the namespace fit in the available bytes
```